deployments/deploy-rancher.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash -x
+
+if [ -n "${VERBOSE}" ]; then
+    VERBOSE="--v"
+fi
+
+if [ -z "${SCALE_FACTOR}" ]; then
+    SCALE_FACTOR="10"
+fi
+
+if [[ -z "$FAIL_FAST" || "$FAIL_FAST" = true ]]; then
+    FAIL_FAST="--failFast"
+else
+    FAIL_FAST="-keepGoing"
+fi
+
+SKIP_ARG=""
+if [ -n "$SKIP_TESTS" ]; then
+    skipRegex=$(echo $SKIP_TESTS | sed -e 's/,/}|{/g')
+    SKIP_ARG="--skip={$skipRegex}"
+fi
+
+FOCUS_ARG=""
+if [ -n "$FOCUS_TESTS" ]; then
+    focusRegex=$(echo $FOCUS_TESTS | sed -e 's/,/}|{/g')
+    FOCUS_ARG="--focus={$focusRegex}"
+fi
+
+UPGRADE_VERSION_ARG=""
+if [ -n "${STORAGE_UPGRADE_VERSION}" ]; then
+    UPGRADE_VERSION_ARG="--storage-driver-upgrade-version=$STORAGE_UPGRADE_VERSION"
+fi
+
+UPGRADE_BASE_VERSION_ARG=""
+if [ -n "${STORAGE_BASE_VERSION}" ]; then
+    UPGRADE_BASE_VERSION_ARG="--storage-driver-base-version=$STORAGE_BASE_VERSION"
+fi
+
+if [ -z "${TORPEDO_IMG}" ]; then
+    TORPEDO_IMG="portworx/torpedo:latest"
+    echo "Using default torpedo image: ${TORPEDO_IMG}"
+fi
+
+kubectl delete pod torpedo
+state=`kubectl get pod torpedo | grep -v NAME | awk '{print $3}'`
+timeout=0
+while [ "$state" == "Terminating" -a $timeout -le 600 ]; do
+  echo "Terminating torpedo..."
+  sleep 1
+  state=`kubectl get pod torpedo | grep -v NAME | awk '{print $3}'`
+  timeout=$[$timeout+1]
+done
+
+if [ $timeout -gt 600 ]; then
+  echo "Torpedo is taking too long to terminate. Operation timeout."
+  describe_pod_then_exit
+fi
+
+TORPEDO_SSH_KEY_VOLUME=""
+TORPEDO_SSH_KEY_MOUNT=""
+if [ -n "${TORPEDO_SSH_KEY}" ]; then
+    kubectl create secret generic key4torpedo --from-file=${TORPEDO_SSH_KEY}
+    TORPEDO_SSH_KEY_VOLUME="{ \"name\": \"ssh-key-volume\", \"secret\": { \"secretName\": \"key4torpedo\", \"defaultMode\": 256 }}"
+    TORPEDO_SSH_KEY_MOUNT="{ \"name\": \"ssh-key-volume\", \"mountPath\": \"/home/torpedo/\" }"
+fi
+
+echo "Deploying torpedo pod..."
+cat <<EOF | kubectl create -f -
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: torpedo-account
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+   name: torpedo-role
+rules:
+  -
+    apiGroups:
+      # have access to everything except Secrets
+      - "*"
+    resources: ["*"]
+    verbs: ["*"]
+  - nonResourceURLs: ["*"]
+    verbs: ["*"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: torpedo-role-binding
+subjects:
+- kind: ServiceAccount
+  name: torpedo-account
+  namespace: default
+roleRef:
+  kind: ClusterRole
+  name: torpedo-role
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: torpedo
+  labels:
+    app: torpedo
+spec:
+  tolerations:
+  - key: node-role.kubernetes.io/controlplane
+    operator: Equal
+    value: "true"
+  - key: node-role.kubernetes.io/etcd
```

Comment:
@thiguetta as discussed yesterday on slack, parameterize the keys in tolerations and the affinity sections and merge into single deploy script. 

We will have many distros of k8s being added to torpedo in near future so we need to avoid adding new scripts wherever some parametrization is possible.